interpreter/cling/lib/Interpreter/IncrementalParser.cpp

```diff
@@ -392,6 +392,28 @@ namespace cling {
            && (!initialized || (m_TransactionPool && m_Parser));
   }
 
+  namespace {
+    template <class T>
+    struct Reverse {
+      const T &m_orig;
+      auto begin() -> decltype(m_orig.rbegin()) { return m_orig.rbegin(); }
+      auto end() -> decltype (m_orig.rend()) { return m_orig.rend(); }
+    };
+    template <class T>
+    Reverse<T> revert(const T& orig) { return {orig}; }
+  }
+
+  const Transaction* IncrementalParser::getLastWrapperTransaction() const {
+    if (auto *T = getCurrentTransaction())
```

Comment:
Would it make sense to cache this value (by setting it in CommitTransaction) rather than searching?

Reply:
We +/- never need this, and it's generally the last transaction anyway. Given this I find caching too expensive.